### PR TITLE
Fix: Buggy pointer magic in copying type context

### PR DESCRIPTION
**Pull request by JuliaPoo**

### CODE REVIEW

File: Python/tier2.c [hunk 1]
```diff
@@ -113,12 +113,17 @@ _PyTier2TypeContext_Copy(const _PyTier2TypeContext *type_context)
             _Py_TYPENODE_t *parent = (_Py_TYPENODE_t *)_Py_TYPENODE_CLEAR_TAG(node);
 
             // Check if part of locals
-            if (parent - type_context->type_locals < nlocals) {
-                type_locals[i] = node - (uintptr_t)orig_type_locals + (uintptr_t)type_locals;
+            int plocals = (int)(parent - type_context->type_locals);
```
Comment (Fidget-Spinner) [inline Python/tier2.c]: Could we use another name please? `plocals` signals its a pointer but it's not, it's just offset. So `offset_locals` maybe?

File: Python/tier2.c [hunk 2]
```diff
@@ -113,12 +113,17 @@ _PyTier2TypeContext_Copy(const _PyTier2TypeContext *type_context)
...
+            if (0 <= plocals && plocals < nlocals) {
+                type_locals[i] = _Py_TYPENODE_MAKE_REF((_Py_TYPENODE_t)(&type_locals[plocals]));
             }
             // Is part of stack
             else {
-                type_locals[i] = node - (uintptr_t)orig_type_stack + (uintptr_t)type_stack;
+#if TYPEPROP_DEBUG
+                int pstack = (int)(parent - type_context->type_stack);
```
Comment (Fidget-Spinner) [inline Python/tier2.c]: Likewise for `pstack`.

**Comment by JuliaPoo** [comment]
Variables are renamed

**Comment by Fidget-Spinner** [comment]
Thanks!

**Pull request merged**